# Genomics

[Whole genome resequencing](experimental-design-genomics.md) was carried out on ~400 sorghum accessions to understand the landscape of genetic variation in the selected germplasm and enable high-resolution mapping of bioenergy traits with genome wide association studies \(GWAS\). Additionally, ~200 sorghum recombinant inbred lines \(RILs\) were characterized with ~400,000 genetic markers using genotyping-by-sequencing \([Morris et al., 2013](http://www.g3journal.org/content/early/2013/09/13/g3.113.008417)\) for trait dissection in the RIL population and testcross hybrids of the RIL population.

## Whole-genome resequencing

**Experimental Design:**

* 384 BAP samples were sequenced to an average depth of ~25x.
  * [List of BAP accessions in BETYdb](https://terraref.org/bety/api/v1/cultivars?ecotype=BAP)
* Shotgun sequencing \(127-bp paired-end\) was done using an Illumina X10 instrument at the HudsonAlpha Institute for Biotechnology.
* Variant calling was done using a [computational pipeline](../protocols/genomic-data.md) at the Danforth Center. 
* Data were aligned against the BTx623 reference genotype
  * NCBI [https://www.ncbi.nlm.nih.gov/assembly/GCF\_000003195.3/](https://www.ncbi.nlm.nih.gov/assembly/GCF_000003195.3/)
  * Phytozome [https://phytozome.jgi.doe.gov/pz/portal.html\#!info?alias=Org\_Sbicolor](https://phytozome.jgi.doe.gov/pz/portal.html#!info?alias=Org_Sbicolor)

## Genotyping-by-sequencing

**Experimental Design:**

* [768 RIL samples](https://terraref.org/bety/api/v1/cultivars?ecotype=RIL) were sequenced using a GBS approach.

## See Also

* [Genomics Protocols](../user-manual/data-products/genomics-data.md)
* [Genomics Data Availability](../user-manual/data-products/genomics-data.md)

### 

